src/metabase/sync.clj

```diff
@@ -49,10 +65,9 @@
     {:keys [scan], :or {scan :full}} :- [:maybe [:map
                                                  [:scan {:optional true} [:maybe [:enum :schema :full]]]]]]
    (sync-util/sync-operation :sync database (format "Sync %s" (sync-util/name-for-logging database))
-     (cond-> [(assoc (sync-metadata/sync-db-metadata! database) :name "metadata")]
-       (= scan :full)
-       (conj (assoc (analyze/analyze-db! database) :name "analyze")
-             (assoc (field-values/update-field-values! database) :name "field-values"))))))
+     (->> (scan-phases scan)
```

Comment:
What if instead we wrap all of these phases with a `(sync-util/run-sync-operation "sync-database" database phases)`?

just like how we do it with [sync-db-metadata!](https://github.com/metabase/metabase/blob/468230746c7ea0021f96bc41bc008d11f2e03b45/src/metabase/sync/sync_metadata.clj#L65)

This will make run each phase with `run-step-with-metadata`, which does it own error handling .

Another benefit is we will now have a `sync-database` task history for all of these phases.

Currently when you call `sync-database!`, we can have task history for `metadata, analyze, field-values` for each step, there are no "parent" task history for all of them.